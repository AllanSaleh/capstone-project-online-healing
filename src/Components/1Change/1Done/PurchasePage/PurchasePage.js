import React, { useState } from 'react';

import Next from './Next.svg';
import Prev from './Prev.svg';
import CreditCard from '../../CreditCard/CreditCard';

let count = 0;

export default function PurchasePage() {
  const cards = [
    {
      date: '04/2025',
      number: '1234 1234 1234 1234',
      name: 'John Doe',
      type: 'Master',
    },
    {
      date: '04/2025',
      number: '1234 1234 1234 1234',
      name: 'John Doe',
      type: 'Visa',
    },
    {
      date: '04/2025',
      number: '1234 1234 1234 1234',
      name: 'John Doe',
      type: 'Master',
    },
    {
      date: '04/2025',
      number: '1234 1234 1234 1234',
      name: 'John Doe',
      type: 'Visa',
    },
    {
      date: '04/2025',
      number: '1234 1234 1234 1234',
      name: 'John Doe',
      type: 'Master',
    },
    {
      date: '04/2025',
      number: '1234 1234 1234 1234',
      name: 'John Doe',
      type: 'Master',
    },
  ];

  const [selected, setSelected] = useState('');

  return (
    <>
      <div className="flex flex-col justify-evenly h-firstsection pt-navbar px-sides">
        <div>
          <div className="text-title">SELECT CARD</div>
          <div className="text-subtitle text-gray-700">
            Please select the card you want to buy the tickets with
          </div>
        </div>

        <div className="flex justify-between items-center">
          <button
            onClick={() => {
              const NewPos = document.getElementById('Menu').scrollLeft - 560;
              document.getElementById('Menu').scrollTo({
                left: NewPos,
                behavior: 'smooth',
              });
            }}
            type="button"
            className="mr-4 h-12 w-12 px-3 text-subtitle bg-white rounded-lg border-2 border-black hover:bg-blue-dark"
          >
            <img src={Prev} alt="Previous" />
          </button>

          <div className="relative h-80 w-full overflow-hidden">
            <div
              id="Menu"
              className="h-96 box-border whitespace-nowrap overflow-x-auto overflow-y-hidden"
            >
              {cards.map((card, i) => {
                count += 1;
                if (count === 4) count = 1;

                return (
                  <CreditCard
                    deletable={false}
                    select={parseInt(selected) === i}
                    setID={(ID) => {
                      if (ID !== selected) setSelected(ID);
                    }}
                    id={i}
                    background={count}
                    logo={card.type}
                    date={card.date}
                    number={card.number}
                    name={card.name}
                  />
                );
              })}
            </div>
          </div>

          <button
            onClick={() => {
              const NewPos = document.getElementById('Menu').scrollLeft + 560;
              document.getElementById('Menu').scrollTo({
                left: NewPos,
                behavior: 'smooth',
              });
            }}
            type="button"
            className="ml-4 h-12 w-12 px-3 text-subtitle bg-white rounded-lg border-2 border-black hover:bg-blue-dark"
          >
            <img src={Next} alt="Previous" />
          </button>
        </div>

        <div className="text-center self-center">
          <div className="text-subtitle self-center">
            Click confirm to use the selected card to purchase 5 tickets for 10$
          </div>
          <button
            onClick={() => {
              if (selected === '') alert('Please choose a Credit Card!');
              else alert(`Used Card ${selected}, Purchase Successful!`);
            }}
            type="button"
            className="mt-8 w-72 h-12 text-subtitle bg-blue-dark rounded-lg border-2 border-transparent hover:bg-white hover:text-blue-dark hover:border-blue-dark"
          >
            CONFIRM PURCHASE
          </button>
        </div>
      </div>
    </>
  );
}
