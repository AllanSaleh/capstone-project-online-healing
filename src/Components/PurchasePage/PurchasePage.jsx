import React, { useState } from 'react';
import { useLocation, useHistory } from 'react-router-dom';

import Next from './Images/Next.svg';
import Prev from './Images/Prev.svg';
import CreditCard from '../CreditCard/CreditCard';

let count = 0;

export default function PurchasePage() {
  const location = useLocation();
  const history = useHistory();

  const cards = [
    {
      date: '04/25',
      number: '1234 1234 1234 1234',
      name: 'John Doe',
      type: 'Master',
    },
    {
      date: '04/25',
      number: '1234 1234 1234 1234',
      name: 'John Doe',
      type: 'Visa',
    },
    {
      date: '04/25',
      number: '1234 1234 1234 1234',
      name: 'John Doe',
      type: 'Master',
    },
    {
      date: '04/25',
      number: '1234 1234 1234 1234',
      name: 'John Doe',
      type: 'Visa',
    },
    {
      date: '04/25',
      number: '1234 1234 1234 1234',
      name: 'John Doe',
      type: 'Master',
    },
    {
      date: '04/25',
      number: '1234 1234 1234 1234',
      name: 'John Doe',
      type: 'Visa',
    },
  ];

  const [selected, setSelected] = useState('');

  const GoToThankYou = () =>
    history.push({
      pathname: '/ThankYou',
      state: 'You purchase has been submitted, you should receive an email with the receipt soon.',
    });

  return (
    <div className="flex flex-col justify-evenly h-firstsection pt-navbar px-sides">
      <div>
        <div className="text-3xl lg:text-title">SELECT CARD</div>
        <div className="text-xl lg:text-subtitle text-gray-700">
          Please select the card you want to buy the tickets with
        </div>
      </div>

      <div className="flex justify-between items-center">
        <button
          onClick={() => {
            const NewPos = document.getElementById('Menu').scrollLeft - 500;
            document.getElementById('Menu').scrollTo({
              left: NewPos,
              behavior: 'smooth',
            });
          }}
          type="button"
          className="mr-2 lg:mr-4 h-10 lg:h-12 w-10 lg:w-12 px-3 bg-white rounded-lg border-2 border-black hover:bg-blue-dark"
        >
          <img src={Prev} alt="Previous" />
        </button>

        <div className="relative h-40 lg:h-80 w-full overflow-hidden">
          <div
            id="Menu"
            className="h-44 lg:h-96 box-border whitespace-nowrap overflow-x-auto overflow-y-hidden"
          >
            {cards.map((card, i) => {
              count += 1;
              if (count === 4) count = 1;

              return (
                <CreditCard
                  deletable={false}
                  select={parseInt(selected) === i}
                  setID={(ID) => {
                    if (ID !== selected) setSelected(ID);
                  }}
                  id={i}
                  background={count}
                  logo={card.type}
                  date={card.date}
                  number={card.number}
                  name={card.name}
                />
              );
            })}
          </div>
        </div>

        <button
          onClick={() => {
            const NewPos = document.getElementById('Menu').scrollLeft + 500;
            document.getElementById('Menu').scrollTo({
              left: NewPos,
              behavior: 'smooth',
            });
          }}
          type="button"
          className="ml-2 lg:ml-4 h-10 lg:h-12 w-10 lg:w-12 px-3 bg-white rounded-lg border-2 border-black hover:bg-blue-dark"
        >
          <img src={Next} alt="Previous" />
        </button>
      </div>

      <div className="text-center self-center">
        <div className="text-xl lg:text-subtitle self-center">
          Click confirm to use the selected card to purchase
          {location.state.ticket} tickets for {location.state.price}$
        </div>

        <button
          onClick={() => {
            if (selected === '') alert('Please choose a Credit Card First!');
            else GoToThankYou();
          }}
          type="button"
          className="mt-8 w-52 lg:w-72 h-12 text-lg lg:text-subtitle bg-blue-dark rounded-lg border-2 border-transparent hover:bg-white hover:text-blue-dark hover:border-blue-dark"
        >
          CONFIRM PURCHASE
        </button>
      </div>
    </div>
  );
}
