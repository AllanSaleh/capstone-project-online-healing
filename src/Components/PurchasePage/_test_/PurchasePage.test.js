import * as React from 'react';
import PurchasePage from "../PurchasePage";
import * as renderer from "react-test-renderer";

it('Purchase Page renders correctly', () => {
  const tree = renderer
    .create(<PurchasePage/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});