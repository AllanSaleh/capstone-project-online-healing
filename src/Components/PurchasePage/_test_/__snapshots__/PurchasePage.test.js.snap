// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="flex flex-col justify-evenly h-firstsection pt-navbar px-sides"
>
  <div>
    <div
      className="text-3xl lg:text-title"
    >
      SELECT CARD
    </div>
    <div
      className="text-xl lg:text-subtitle text-gray-700"
    >
      Please select the card you want to buy the tickets with
    </div>
  </div>
  <div
    className="flex justify-between items-center"
  >
    <button
      className="mr-2 lg:mr-4 h-10 lg:h-12 w-10 lg:w-12 px-3 bg-white rounded-lg border-2 border-black hover:bg-blue-dark"
      onClick={[Function]}
      type="button"
    >
      <img
        alt="Previous"
        src="Prev.svg"
      />
    </button>
    <div
      className="relative h-40 lg:h-80 w-full overflow-hidden"
    >
      <div
        className="h-44 lg:h-96 box-border whitespace-nowrap overflow-x-auto overflow-y-hidden"
        id="Menu"
      >
        <div
          className="relative inline-block w-full lg:w-1/3 h-40 lg:h-80 box-border mx-2 lg:mx-12"
          id={0}
          onClick={[Function]}
        >
          <img
            alt="Background"
            className="border-8 rounded-lg border-dashed absolute top-4 lg:top-0 w-full cursor-pointer border-transparent"
            src="C1.svg"
          />
          <img
            alt="Logo"
            className="w-8 lg:w-auto cursor-pointer absolute top-8 right-4 lg:top-11 lg:right-6"
            src="MasterLogo.svg"
          />
          <div
            className="cursor-pointer absolute top-8 left-4 lg:top-12 lg:left-12 text-white text-sm lg:text-paragraph"
          >
            04/2025
          </div>
          <div
            className="cursor-pointer absolute top-16 left-4 lg:top-32 lg:left-12 text-white text-sm lg:text-paragraph"
          >
            1234 1234 1234 1234
          </div>
          <div
            className="cursor-pointer absolute top-24 left-4 lg:top-52 lg:left-12 text-white text-sm lg:text-paragraph"
          >
            John Doe
          </div>
          <button
            className="absolute bottom-7 right-4 h-6 w-12 lg:bottom-9 lg:right-4 lg:h-12 lg:w-28 text-xs lg:text-subtitle text-white bg-blue-dark rounded-lg border-2 border-transparent hover:bg-white hover:text-blue-dark hover:border-blue-dark"
            type="button"
          >
            DELETE
          </button>
        </div>
        <div
          className="relative inline-block w-full lg:w-1/3 h-40 lg:h-80 box-border mx-2 lg:mx-12"
          id={1}
          onClick={[Function]}
        >
          <img
            alt="Background"
            className="border-8 rounded-lg border-dashed absolute top-4 lg:top-0 w-full cursor-pointer border-transparent"
            src="C2.svg"
          />
          <img
            alt="Logo"
            className="w-8 lg:w-auto cursor-pointer absolute top-8 right-4 lg:top-11 lg:right-6"
            src="VisaLogo.svg"
          />
          <div
            className="cursor-pointer absolute top-8 left-4 lg:top-12 lg:left-12 text-white text-sm lg:text-paragraph"
          >
            04/2025
          </div>
          <div
            className="cursor-pointer absolute top-16 left-4 lg:top-32 lg:left-12 text-white text-sm lg:text-paragraph"
          >
            1234 1234 1234 1234
          </div>
          <div
            className="cursor-pointer absolute top-24 left-4 lg:top-52 lg:left-12 text-white text-sm lg:text-paragraph"
          >
            John Doe
          </div>
          <button
            className="absolute bottom-7 right-4 h-6 w-12 lg:bottom-9 lg:right-4 lg:h-12 lg:w-28 text-xs lg:text-subtitle text-white bg-blue-dark rounded-lg border-2 border-transparent hover:bg-white hover:text-blue-dark hover:border-blue-dark"
            type="button"
          >
            DELETE
          </button>
        </div>
        <div
          className="relative inline-block w-full lg:w-1/3 h-40 lg:h-80 box-border mx-2 lg:mx-12"
          id={2}
          onClick={[Function]}
        >
          <img
            alt="Background"
            className="border-8 rounded-lg border-dashed absolute top-4 lg:top-0 w-full cursor-pointer border-transparent"
            src="C3.svg"
          />
          <img
            alt="Logo"
            className="w-8 lg:w-auto cursor-pointer absolute top-8 right-4 lg:top-11 lg:right-6"
            src="MasterLogo.svg"
          />
          <div
            className="cursor-pointer absolute top-8 left-4 lg:top-12 lg:left-12 text-white text-sm lg:text-paragraph"
          >
            04/2025
          </div>
          <div
            className="cursor-pointer absolute top-16 left-4 lg:top-32 lg:left-12 text-white text-sm lg:text-paragraph"
          >
            1234 1234 1234 1234
          </div>
          <div
            className="cursor-pointer absolute top-24 left-4 lg:top-52 lg:left-12 text-white text-sm lg:text-paragraph"
          >
            John Doe
          </div>
          <button
            className="absolute bottom-7 right-4 h-6 w-12 lg:bottom-9 lg:right-4 lg:h-12 lg:w-28 text-xs lg:text-subtitle text-white bg-blue-dark rounded-lg border-2 border-transparent hover:bg-white hover:text-blue-dark hover:border-blue-dark"
            type="button"
          >
            DELETE
          </button>
        </div>
        <div
          className="relative inline-block w-full lg:w-1/3 h-40 lg:h-80 box-border mx-2 lg:mx-12"
          id={3}
          onClick={[Function]}
        >
          <img
            alt="Background"
            className="border-8 rounded-lg border-dashed absolute top-4 lg:top-0 w-full cursor-pointer border-transparent"
            src="C1.svg"
          />
          <img
            alt="Logo"
            className="w-8 lg:w-auto cursor-pointer absolute top-8 right-4 lg:top-11 lg:right-6"
            src="VisaLogo.svg"
          />
          <div
            className="cursor-pointer absolute top-8 left-4 lg:top-12 lg:left-12 text-white text-sm lg:text-paragraph"
          >
            04/2025
          </div>
          <div
            className="cursor-pointer absolute top-16 left-4 lg:top-32 lg:left-12 text-white text-sm lg:text-paragraph"
          >
            1234 1234 1234 1234
          </div>
          <div
            className="cursor-pointer absolute top-24 left-4 lg:top-52 lg:left-12 text-white text-sm lg:text-paragraph"
          >
            John Doe
          </div>
          <button
            className="absolute bottom-7 right-4 h-6 w-12 lg:bottom-9 lg:right-4 lg:h-12 lg:w-28 text-xs lg:text-subtitle text-white bg-blue-dark rounded-lg border-2 border-transparent hover:bg-white hover:text-blue-dark hover:border-blue-dark"
            type="button"
          >
            DELETE
          </button>
        </div>
        <div
          className="relative inline-block w-full lg:w-1/3 h-40 lg:h-80 box-border mx-2 lg:mx-12"
          id={4}
          onClick={[Function]}
        >
          <img
            alt="Background"
            className="border-8 rounded-lg border-dashed absolute top-4 lg:top-0 w-full cursor-pointer border-transparent"
            src="C2.svg"
          />
          <img
            alt="Logo"
            className="w-8 lg:w-auto cursor-pointer absolute top-8 right-4 lg:top-11 lg:right-6"
            src="MasterLogo.svg"
          />
          <div
            className="cursor-pointer absolute top-8 left-4 lg:top-12 lg:left-12 text-white text-sm lg:text-paragraph"
          >
            04/2025
          </div>
          <div
            className="cursor-pointer absolute top-16 left-4 lg:top-32 lg:left-12 text-white text-sm lg:text-paragraph"
          >
            1234 1234 1234 1234
          </div>
          <div
            className="cursor-pointer absolute top-24 left-4 lg:top-52 lg:left-12 text-white text-sm lg:text-paragraph"
          >
            John Doe
          </div>
          <button
            className="absolute bottom-7 right-4 h-6 w-12 lg:bottom-9 lg:right-4 lg:h-12 lg:w-28 text-xs lg:text-subtitle text-white bg-blue-dark rounded-lg border-2 border-transparent hover:bg-white hover:text-blue-dark hover:border-blue-dark"
            type="button"
          >
            DELETE
          </button>
        </div>
        <div
          className="relative inline-block w-full lg:w-1/3 h-40 lg:h-80 box-border mx-2 lg:mx-12"
          id={5}
          onClick={[Function]}
        >
          <img
            alt="Background"
            className="border-8 rounded-lg border-dashed absolute top-4 lg:top-0 w-full cursor-pointer border-transparent"
            src="C3.svg"
          />
          <img
            alt="Logo"
            className="w-8 lg:w-auto cursor-pointer absolute top-8 right-4 lg:top-11 lg:right-6"
            src="VisaLogo.svg"
          />
          <div
            className="cursor-pointer absolute top-8 left-4 lg:top-12 lg:left-12 text-white text-sm lg:text-paragraph"
          >
            04/2025
          </div>
          <div
            className="cursor-pointer absolute top-16 left-4 lg:top-32 lg:left-12 text-white text-sm lg:text-paragraph"
          >
            1234 1234 1234 1234
          </div>
          <div
            className="cursor-pointer absolute top-24 left-4 lg:top-52 lg:left-12 text-white text-sm lg:text-paragraph"
          >
            John Doe
          </div>
          <button
            className="absolute bottom-7 right-4 h-6 w-12 lg:bottom-9 lg:right-4 lg:h-12 lg:w-28 text-xs lg:text-subtitle text-white bg-blue-dark rounded-lg border-2 border-transparent hover:bg-white hover:text-blue-dark hover:border-blue-dark"
            type="button"
          >
            DELETE
          </button>
        </div>
      </div>
    </div>
    <button
      className="ml-2 lg:ml-4 h-10 lg:h-12 w-10 lg:w-12 px-3 bg-white rounded-lg border-2 border-black hover:bg-blue-dark"
      onClick={[Function]}
      type="button"
    >
      <img
        alt="Previous"
        src="Next.svg"
      />
    </button>
  </div>
  <div
    className="text-center self-center"
  >
    <div
      className="text-xl lg:text-subtitle self-center"
    >
      Click confirm to use the selected card to purchase 5 tickets for 10$
    </div>
    <button
      className="mt-8 w-52 lg:w-72 h-12 text-lg lg:text-subtitle bg-blue-dark rounded-lg border-2 border-transparent hover:bg-white hover:text-blue-dark hover:border-blue-dark"
      onClick={[Function]}
      type="button"
    >
      CONFIRM PURCHASE
    </button>
  </div>
</div>
`;
